### PR TITLE
Add `util.IsBinaryModuleInstalled`

### PR DESCRIPTION
**Pull request by WilliamVenner**
Improvement on #1868 (sorry Down nothing personal just wanted a separate PR) with edge case fix for Linux 32-bit x86-64 branch and a more code golf form of generating the format string of the module path. Also I changed the name to make it a bit more explicit.

### CODE REVIEW

File: garrysmod/lua/includes/extensions/util.lua [hunk 1]
```diff
@@ -365,3 +365,33 @@ function util.RemovePData( steamid, name )
 	sql.Query( "DELETE FROM playerpdata WHERE infoid = " .. SQLStr( name ) )
 
 end
+
+--[[---------------------------------------------------------
+	Name: BinaryModuleInstalled( name )
+	Desc: Returns whether a binary module with the given name is present on disk
+-----------------------------------------------------------]]
+do
```
Comment (robotboy655) [inline garrysmod/lua/includes/extensions/util.lua:373]: What is this for?

File: garrysmod/lua/includes/extensions/util.lua [hunk 2]
```diff
@@ -365,3 +365,33 @@ function util.RemovePData( steamid, name )
...
+-----------------------------------------------------------]]
+do
+	local suffix = ({"osx64","osx","linux64","linux","win64","win32"})[
+		( system.IsWindows() && 4 || 0 )
+		+ ( system.IsLinux() && 2 || 0 )
+		+ ( jit.arch == "x86" && 1 || 0 )
+		+ 1
+	]
+	local fmt = "lua/bin/gm" .. (CLIENT && "cl" || "sv") .. "_%s_%s.dll"
+	function util.BinaryModuleInstalled( name )
```
Comment (robotboy655) [inline garrysmod/lua/includes/extensions/util.lua:381]: Shouldn't it be `util.IsBinaryModuleInstalled`?

File: garrysmod/lua/includes/extensions/util.lua [hunk 3]
```diff
@@ -365,3 +365,33 @@ function util.RemovePData( steamid, name )
...
+		if ( !isstring( name ) || #name == 0 ) then
+			return false
```
Comment (robotboy655) [inline garrysmod/lua/includes/extensions/util.lua:383]: We shouldn't silently fail on invalid input, it should error.

**Review by robotboy655 — CHANGES_REQUESTED** [review]

File: garrysmod/lua/includes/extensions/util.lua (hunk 1 above)
Comment (WilliamVenner) [reply garrysmod/lua/includes/extensions/util.lua:373]: Scope encapsulation. Just separates `suffix` and `fmt` from the rest of the code, it's not necessary, just a Lua pattern

File: garrysmod/lua/includes/extensions/util.lua (hunk 2 above)
Comment (WilliamVenner) [reply garrysmod/lua/includes/extensions/util.lua:381]: Yes... it should 🤔

File: garrysmod/lua/includes/extensions/util.lua (hunk 1 above)
Comment (robotboy655) [reply garrysmod/lua/includes/extensions/util.lua:373]: I get that, my point is that nothing else in the codebase uses this.

**Review by robotboy655 — APPROVED** [review]

**Pull request merged**